### ISSUE TITLE
Syntax Error: expected expression

### ISSUE THREAD

**Issue by jyn514** [rust-analyzer/rust-analyzer#2801]
I get incorrect errors on the following code. I think rust-analyzer might not like directives in the middle of a function call.

```rust
fn assert_float(s: &str, n: f64) {}

fn main() {
    assert_float(
        "1.797693134862315708e+308L",
        #[allow(clippy::excessive_precision)]
        1.797_693_134_862_315_730_8e+308,
    );
}
```

shows up right after `+308L",`:
> Syntax Error: expected expression
> Syntax Error: expected SEMI

The errors also show up after `e+308,`.

**Comment by jyn514** [comment]
Update: removing the `clippy` directive makes the errors go away.

**Comment by bjorn3** [comment]
Does this happen for all attributes in that position or only clippy lint levels?

**Comment by jyn514** [comment]
All attributes in that position. I tested with ` #[allow(dead_code)]`.

**Comment by lnicola** [comment]
Same as https://github.com/rust-analyzer/rust-analyzer/issues/2781?

**Comment by bjorn3** [comment]
#2781 is for tuples while this is for call params. They are very similar though.

**Comment by edwin0cheng** [comment]
Yeah, we didn't handle Attr in arguments list yet :

https://github.com/rust-analyzer/rust-analyzer/blob/86d2af9f7b3f6af2ea9e4eca2584aa501b60aa14/crates/ra_parser/src/grammar/expressions.rs#L538-L554

**Comment by jyn514** [comment]
I can take a stab at this :) I saw there were instructions in https://github.com/rust-analyzer/rust-analyzer/tree/master/docs/dev, I'll try and make a start today.
